src/ir_Corona.cpp

```diff
@@ -0,0 +1,569 @@
+  return GETBITS8(remote_state[kCoronaAcSectionD1Pos], kCoronaAcTempOffset,
+      kCoronaAcTempSize) + kCoronaAcMinTemp - 1;
+}
+
+/// Change the power setting.
+/// @param on true, the setting is on. false, the setting is off.
+/// @note If changed, setPowerToggle is also needed,
+///       unless timer is or was active
+void IRCoronaAc::setPower(const bool on) {
+  setBit(&remote_state[kCoronaAcSectionD1Pos], kCoronaAcPowerOffset, on);
+  // setting power state resets timers that would cause the state
+  if (on)
+    setOnTimer(kCoronaAcTimerOff);
+  else
+    setOffTimer(kCoronaAcTimerOff);
+}
+
+/// Get the value of the current power setting.
+/// @return true, the setting is on. false, the setting is off.
+bool IRCoronaAc::getPower(void) {
+  return GETBIT8(remote_state[kCoronaAcSectionD1Pos], kCoronaAcPowerOffset);
+}
+
+/// Change the power toggle setting.
+/// @param on true, the setting is on. false, the setting is off.
+/// @note this sets that the AC should change power,
+///       use setPower to define if the AC should end up as on or off
+void IRCoronaAc::setPowerToggle(const bool toggle) {
+  setBit(&remote_state[kCoronaAcSectionD1Pos],
+      kCoronaAcPowerToggleOffset, toggle);
+}
+
+/// Get the value of the current power toggle setting.
+/// @return true, the setting is on. false, the setting is off.
+bool IRCoronaAc::getPowerToggle(void) {
+  return GETBIT8(remote_state[kCoronaAcSectionD1Pos],
+      kCoronaAcPowerToggleOffset);
+}
+
+/// Change the power setting to On.
+void IRCoronaAc::on(void) { setPower(true); }
+
+/// Change the power setting to Off.
+void IRCoronaAc::off(void) { setPower(false); }
+
+/// Get the operating mode setting of the A/C.
+/// @return The current operating mode setting.
+uint8_t IRCoronaAc::getMode(void) {
+  return GETBITS8(remote_state[kCoronaAcSectionD1Pos],
+      kCoronaAcModeOffset, kCoronaAcModeSize);
+}
+
+/// Set the operating mode of the A/C.
+/// @param mode The desired operating mode.
+void IRCoronaAc::setMode(const uint8_t mode) {
+  switch (mode) {
+    case kCoronaAcModeCool:
+    case kCoronaAcModeDry:
+    case kCoronaAcModeFan:
+    case kCoronaAcModeHeat:
+      setBits(&remote_state[kCoronaAcSectionD1Pos],
+              kCoronaAcModeOffset, kCoronaAcModeSize,
+              mode);
+      return;
+    default:
+      this->setMode(kCoronaAcModeCool);
+  }
+}
+
+/// Convert a standard A/C mode into its native mode.
+/// @param mode A stdAc::opmode_t mode to be converted to it's native equivalent
+/// @return The corresponding native mode.
+uint8_t IRCoronaAc::convertMode(const stdAc::opmode_t mode) {
+  switch (mode) {
+    case stdAc::opmode_t::kFan:  return kCoronaAcModeFan;
+    case stdAc::opmode_t::kDry:  return kCoronaAcModeDry;
+    case stdAc::opmode_t::kHeat: return kCoronaAcModeHeat;
+    default: return kCoronaAcModeCool;
+  }
+}
+
+/// Convert a native mode to it's common stdAc::opmode_t equivalent.
+/// @param mode A native operation mode to be converted.
```

Comment:
General `@param` nitpick for here and elsewhere. Please mark them as `@param[in]` etc etc for directionality. (yeah, I missed that heaps too. I did a giant search & replace over the entire project just before. ;-)